src/autocomplete.js

```diff
@@ -362,8 +362,6 @@ class Autocomplete {
             if (this.tooltipNode) {
                 this.updateDocTooltip();
             }
-        } else if (keepPopupPosition && !prefix) {
-            this.detach();
```

Comment:
Looks like this was added in https://github.com/ajaxorg/ace/commit/bd95f782d8fd5be28334e8783bba52406d95654c to close the popup when backspacing into an empty string. E.g.:

```
// type something like `con`
con
// press CTRL+Space -> pop-up renders with suggestions like "console.(...)"
// backspace 3 times
<empty line>
// what happens to the popup here after this PR?
```

Reply:
yeah it does change the behavior there

https://github.com/ajaxorg/ace/assets/34573235/b7d049df-2e1a-4691-931d-2feddd821a0c

Reply:
In some sense, backspacing until it becomes an empty prefix is similar to typing until it's an empty prefix, might be hard to keep the old behavior while backspacing while still allowing for empty prefix completion while typing ahead.

Reply:
Maybe we could do something similar to the logic in https://github.com/ajaxorg/ace/blob/6e870fd08e06be47f9da971743d13b25c41f99c2/src/ext/language_tools.js#L140-L153?

I.e., different behaviour depending on whether it's a deletion or an insertion of text

Reply:
Maybe a command for `"Backspace"` close to the other commands: https://github.com/ajaxorg/ace/blob/6e870fd08e06be47f9da971743d13b25c41f99c2/src/autocomplete.js#L738-L757

Something along the lines of "if backspace and no prefix then detach"

Reply:
I went for the second option in a new commit